pkg/extender/extender.go

```diff
@@ -847,3 +919,168 @@ func (e *Extender) processCSIExtPodPrioritizeRequest(
 		storklog.PodLog(pod).Errorf("Failed to encode response: %v", err)
 	}
 }
+
+func (e *Extender) processVirtLauncherPodPrioritizeRequest(
+	encoder *json.Encoder,
+	args schedulerapi.ExtenderArgs) error {
+	pod := args.Pod
+
+	vmiName, vmiUID := e.getVMIInfo(pod)
+	if vmiName == "" {
+		return fmt.Errorf("unable to find vmiName for pod: %v", pod.Name)
+	}
+	podBeingLiveMigrated, err := e.getLiveMigrationInfo(pod, vmiUID)
+	if err != nil {
+		return fmt.Errorf("unable to find liveMigrationInfo: %w", err)
+	}
+
+	if podBeingLiveMigrated != nil {
+		storklog.PodLog(pod).Infof("Live VM migration in progress: (vmiName: %v, podBeingLiveMigrated: %v)", vmiName, podBeingLiveMigrated.Name)
+	} else {
+		storklog.PodLog(pod).Infof("No ongoing live VM migration in progress. vmiName: %v", vmiName)
+	}
+
+	vmi, err := kvd.Instance().GetVirtualMachineInstance(context.TODO(), pod.Namespace, vmiName)
+	if err != nil {
+		return fmt.Errorf("unable to find VMI Info: %w", err)
+	}
+	storklog.PodLog(pod).Debugf("VMI is using PVC: %v", vmi.RootDiskPVC)
+
+	pvName, err := e.Driver.GetPVNameFromPVC(vmi.RootDiskPVC, pod.Namespace)
+	if err != nil {
+		return fmt.Errorf("unable to inspect PVC: %v err: %w", vmi.RootDiskPVC, err)
+	}
+	storklog.PodLog(pod).Debugf("PV for scoring: %v", pvName)
+
+	volInfo, err := e.Driver.InspectVolume(pvName)
+	if err != nil {
+		return fmt.Errorf("unable to inspect PV: %v. err: %w", pvName, err)
+	}
+
+	if podBeingLiveMigrated != nil {
+		podBeingLiveMigratedIsBindMounted := e.isUsingBindMount(podBeingLiveMigrated, volInfo)
+
+		if podBeingLiveMigratedIsBindMounted {
+			// Live VM Migration in Progress and existing Pod is bind mounted => Antihyperconvergence
+			// i.e. Give lower scores to nodes with volume replicas
+			storklog.PodLog(pod).Infof("Pod %v is running with same VMI owner reference is using localAttachment.", podBeingLiveMigrated.Name)
+			e.updateVirtLauncherPodPrioritizeScores(encoder,
+				args,
+				volInfo,
+				false, /*preferLocalAttachment*/
+				true /*antihyperconvergence*/)
+		} else {
+			// Live VM Migration in Progress but existing Pod is using NFS using mount
+			// Use Antihyperconvergence to give lower score to replica nodes without volume attachment
```

Comment:
Can you add more clarification here: 
We give lower score to replica nodes so that if a pod is unable to schedule on the node where the volume is attached we can run it on a non replica node, so that a subsequent LiveMigration can move the pod back to the replica node in a single hop.

Reply:
Added the comment you suggested to if (podBeingLiveMigrated) condition as this is valid not only in the else here